gems/smithy/lib/smithy/views/client/protocol_spec.rb

```diff
@@ -0,0 +1,239 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Views
+    module Client
+      # @api private
+      class ProtocolSpec < View
+        PROTOCOL_TEST_TRAITS = %w[smithy.test#httpRequestTests smithy.test#httpResponseTests].freeze
+
+        def initialize(plan)
+          @plan = plan
+          @model = plan.model
+          @all_operation_tests = Model::ServiceIndex
```

Comment:
Nit: operation tests

Reply:
What do you mean?

Reply:
All_ prefix just seemed unnecessary. It's a collection of operation tests.

Reply:
it conflicts with `operation_tests` below.  `operations_tests` felt harder to read (the difference is this is a list of tests for all operations.  operation_tests is the tests for a single operation).

Reply:
I was using it in the view:
```
<% all_operation_tests.each do |operation_tests| %>
```

Not sure of better naming for both of these.

Reply:
I guess you could name them protocol_tests which enumerates operation_tests, but whatever is fine. Is the operation separation relevant or important?

Reply:
Hmm - yeah thats a good question.  

Tests themselves are modeled per operation and typically rspec breaks down to describe '#operation' behaviors.  But theres no fundamental reason we couldn't just generate a full list and skip going operation by operation.  
There might be some complexity in the IDs.  I'm not 100% sure, but I think IDs are unique within request tests and response tests.  See: https://github.com/smithy-lang/smithy/blob/main/smithy-protocol-test-traits/src/main/java/software/amazon/smithy/protocoltests/traits/UniqueProtocolTestCaseIdValidator.java

So I thikn it would be fine as long as we still had seperate `describe 'requests'` and resposne blocks.

Reply:
Are there protocol tests for certain operations like for S3?

Reply:
I recall seeing service-specific protocol tests somewhere - let me find it.

Reply:
https://github.com/smithy-lang/smithy/blob/main/smithy-aws-protocol-tests/model/restXml/services/s3.smithy